src/client/options/test.rs

```diff
@@ -332,3 +332,43 @@ async fn run_uri_options_spec_tests() {
 async fn run_connection_string_spec_tests() {
     run_spec_test(&["connection-string"], run_test).await;
 }
+
+async fn parse_uri(option: &str, suggestion: Option<&str>) {
+    match ClientOptionsParser::parse(&format!("mongodb://host:27017/?{}=test", option))
+        .as_ref()
+        .map_err(|e| e.as_ref())
+    {
+        Ok(_) => panic!("expected error for option {}", option),
+        Err(ErrorKind::ArgumentError { message, .. }) => {
+            match suggestion {
+                Some(s) => assert_eq!(
+                    message,
+                    &format!(
```

Comment:
instead of checking for exact equality here, we might want to just check that the suggestion appears as a substring in the message. That way minor tweaks to this message won't break this test in the future.

For the `None` case, maybe we just pick a word we expect to be there in the future like "similar".

Reply:
"invalid" should appear regardless of whether we suggest something or not, and in the `None` case we want to verify that no suggestion is made. To that end, we may want to assert that the error message does _not_ contain a certain substring (e.g. "similar") that will only appear when something is suggested.

Also, that lint failure is unrelated to your changes and will go away once #164 is merged.